packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -916,7 +916,20 @@ Future<void> _writeMacOSPluginRegistrant(FlutterProject project, List<Plugin> pl
 }
 
 Future<void> _writeWindowsPluginFiles(FlutterProject project, List<Plugin> plugins) async {
-  final List<Map<String, dynamic>> windowsPlugins = _extractPlatformMaps(plugins, WindowsPlugin.kConfigKey);
+  // Plugins without a 'pluginClass' definition shouldn't create native cpp files. The plugin might
+  // be a pure Dart implementation, and the class will be defined by 'dartPluginClass'.
+  final List<Plugin> nativeWindowsPlugins = plugins.where((final Plugin element) {
+    final WindowsPlugin windowsPlugin = element.platforms[WindowsPlugin.kConfigKey] as WindowsPlugin;
+    if (windowsPlugin == null) {
+      return false;
+    }
+    return windowsPlugin.pluginClass != null && windowsPlugin.pluginClass.isNotEmpty;
+  }).toList();
+  if (nativeWindowsPlugins.isEmpty) {
```

Comment:
This is not correct; if it's empty, we need to update the files accordingly, as it might not have previously been empty.

It would be nice to add a test that would have caught this, although since it's not a new issue that we're missing coverage it doesn't need to be part of this PR necessarily.

Reply:
Should we delete the files then? Wouldn't it create registrant files with empty class names, or I think it would actually fail.

Reply:
Nvm, just tested this. It was failing from another issue :)